File: day1/day1.py
with open('./input.txt') as f:
    numbers = [int(line.strip()) for line in f]


def findNums(input) :
    dict = {}

    for num in input:
        m = 2020 - num
        if m in dict:
            return [dict[m], m]
        else:
            dict[num] = m

answer = findNums(numbers)

print(answer[0] * answer[1])

File: day1/day1.js
var fs = require('fs')

const text = fs.readFileSync("./input.txt").toString();
const nums = text.split("\r\n").map(Number);

let dict = {};
const target = 2020;
let answer;

for (let i = 0; i < nums.length; i++) {
    let m = target - nums[i]

    if (dict[m]) {
        answer = ([dict[m], m])
    } else {
        dict[nums[i]] = m
    }
};

console.log(answer[0]* answer[1])